src/state/user/reducer.ts

```diff
@@ -20,13 +21,14 @@ export interface UserState {
 
   userExpertMode: boolean
 
-  userClientSideRouter: boolean // whether routes should be calculated with the client side router only
+  // which router should be used to calculate trades
+  userRouterPreference: RouterPreference
 
   // hides closed (inactive) positions across the app
   userHideClosedPositions: boolean
 
   // user defined slippage tolerance in bips, used in all txns
-  userSlippageTolerance: number | 'auto'
+  userSlippageTolerance: number | SlippageTolerance.Auto
   userSlippageToleranceHasBeenMigratedToAuto: boolean // temporary flag for migration status
```

Comment:
Can this temporary flag be removed?

Reply:
It would be best to follow-up with the original author, that has more context - I am definitely not aware of the migration and how was it performed. I will follow-up with whoever authored this code early next week. Meanwhile, I've added a TODO with Jira ticket.

Reply:
After doing some research, turns out that this flag is unlikely to be removed in the future, unless we're absolutely certain that all users of Uniswap have successfully loaded their app at least once since this feature was added. This flag is used to migrate / hydrate pre-existing localStorage state with the default values required to work. Removing it will make a very very small percentage (but still) of users experience issues, unless they do hard reload.

Reply:
Perhaps in the future, we should be a bit more careful around how we roll out those features and avoid adding new flags/properties as much as possible.